### PR TITLE
A `README` for the preprocessing layers

### LINKED ISSUE #1878: Add Table for bbox and segmentation
**Short Description**

Add a table representing which layer is supports bbox/segmentation in `README.md`

something like:

| Layer Name | Image | Masks | BBoxes |
| ----| :---: | :---: | :----: |
| RandomFlip  | ✓     | ✓     | ✓      |

@ianstenbit @jbischof

### PR DESCRIPTION
**Pull request by ariG23498**
Fixes #1878 

CC: @ianstenbit

### CODE REVIEW

File: keras_cv/layers/preprocessing/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Base Layers
+
+- BaseImageAugmentationLayer
+- VectorizedBaseImageAugmentationLayer
+- RandomAugmentationPipeline
+
+# Preprocessing Layers
+
+| Layer Name | Base Layer | Segmentation Masks | BBoxes |
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/README.md:9]: Let's just make this "vectorized' instead of listing out the base layer

Comment (ianstenbit) [inline keras_cv/layers/preprocessing/README.md:9]: Let's also add a column for "class labels"

File: keras_cv/layers/preprocessing/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Base Layers
+
+- BaseImageAugmentationLayer
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/README.md:3]: I suppose if we're going to include these, we should link to the implementations.


That said, I don't know how critical this is -- we should move it to the bottom of the README

File: keras_cv/layers/preprocessing/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+| Equalization | BaseImageAugmentationLayer | ✅ | ✅ |
+| FourierMix | BaseImageAugmentationLayer | ❌ | ❌ |
+| Grayscale | VectorizedBaseImageAugmentationLayer | ✅ | ✅ |
+| GridMask | BaseImageAugmentationLayer | ✅ | ✅ |
+| JitteredResize | VectorizedBaseImageAugmentationLayer | ✅ | ✅ |
+| MixUp | BaseImageAugmentationLayer | ✅ | ✅ |
+| Mosaic | VectorizedBaseImageAugmentationLayer | ✅ | ✅ |
+| Posterization | BaseImageAugmentationLayer | ✅ | ✅ |
+| RandAugment | RandomAugmentationPipeline | ❌ | ❌ |
+| RandomApply | BaseImageAugmentationLayer | ❌ | ❌ |
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/README.md:24]: Meta layers like Randomapply/RandomChoice support all data types that their sub-layers support. Maybe a checkmark with an asterisk and an explanation below would be best for this one.

File: keras_cv/layers/preprocessing/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Base Layers
+
+- BaseImageAugmentationLayer
+- VectorizedBaseImageAugmentationLayer
+- RandomAugmentationPipeline
+
+# Preprocessing Layers
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/README.md:7]: Let's add a little text about how the augmentation API can be used, what the data i/o format is like, and what the meaning of this table is.

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thank you for this PR! This is great 😄

File: keras_cv/layers/preprocessing/README.md (hunk 2 above)
Comment (ariG23498) [reply keras_cv/layers/preprocessing/README.md:64]: I have just moved it to the bottom of the md file. If you think this is not necessary we can remove it as well.

File: keras_cv/layers/preprocessing/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+| RandomGaussianBlur | ❌ | ✅ | ✅ | ✅ |
+| RandomHue | ✅ | ✅ | ✅ | ✅ |
+| RandomJpegQuality | ❌ | ✅ | ✅ | ✅ |
+| RandomRotation | ✅ | ✅ | ✅ | ✅ |
+| RandomSaturation | ✅ | ✅ | ✅ | ✅ |
+| RandomSharpness | ✅ | ✅ | ✅ | ✅ |
+| RandomShear | ✅ | ❌ | ✅ | ✅ |
+| RandomTranslation | ✅ | ❌ | ✅ | ✅ |
+| RandomZoom | ✅ | ❌ | ❌ | ✅ |
+| RepeatedAugmentation | ❌ | ❌ | ❌ | ❌ |
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/README.md:55]: This is also a meta-layer

File: keras_cv/layers/preprocessing/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+# Preprocessing Layers
+
+KerasCV offers a bunch of pre-processing layers.
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/README.md:3]: KerasCV offers many preprocessing and data augmentation layers which support classification, object detection, and segmentation masks. When you use KerasCV augmentation layers to augment your training data, class labels, bounding boxes, and mask labels automatically get augmented alongside the image augmentations!

**Review by ianstenbit — APPROVED** [review]
LGTM, just a few little cleanup bits/

Thank you!

**Comment by cosmo3769** [comment]
@ariG23498 @ianstenbit 
I would love to add support of segmentation mask in the preprocessing layers where it is missing. Thank you.

**Comment by ariG23498** [comment]
That would be great!

I am currently trying to collect all the layers missing the support in a master issue -- We can maybe continue our conversation there?

CC: @ianstenbit

**Comment by cosmo3769** [comment]
Yeah, that would make tracking the progress easy.
